lib/schema_registry/client.rb

```diff
@@ -27,11 +28,21 @@ def initialize(code, message)
 
   class Client
 
-    attr_reader :endpoint, :username, :password
+    attr_reader :endpoint, :username, :password, :client_key, :client_certificate, :ca_certificate, :verify_mode
 
-    def initialize(endpoint, username = nil, password = nil)
+    def initialize(endpoint, username = nil, password = nil, client_certificate = nil, client_key = nil, ca_certificate = nil, verify_mode = :verify_peer)
```

Comment:
Whoops, I should have been more clear. I was talking about proper keyword arguments, no optional arguments.

``` ruby
  def initialize(endpoint, username = nil, password = nil, client_certificate: nil, client_key: nil, ca_certificate: nil, verify_mode: OpenSSL::SSL::VERIFY_PEER)
```